subprojects/execution/src/main/java/org/gradle/internal/execution/WorkValidationException.java

```diff
@@ -27,6 +27,10 @@
  */
 @Contextual
 public class WorkValidationException extends DefaultMultiCauseException {
```

Comment:
Should this remain a `MultiCauseException`? Wouldn't a `GradleException` be sufficient now?

I also wonder whether this should be a subtype of `InvalidUserCodeException` or `InvalidUserDataException`. Though probably `GradleException` is enough.

Reply:
I didn't touch this because `WorkValidationException` is an old class. I think it's fine to keep use of it as the name carries semantics which are stronger than a generic `GradleException`. It's also used in more contexts so I cannot change its supertype.

Reply:
I only see the class used `ValidateStep` and `ValidatePlugins`, and we should probably change both places and not only `ValidatePlugins`. The exception is also internal, so I don't see a problem in changing the superclass. WDYT?